src/main/java/org/ibp/api/rest/location/LocationResource.java

```diff
@@ -53,20 +61,49 @@
 		return new ResponseEntity<>(this.locationService.getLocationTypes(), HttpStatus.OK);
 	}
 
-
-	@ApiOperation(value = "List locations", notes = "Get a list of locations filter by types")
+	@ApiImplicitParams({
+		@ApiImplicitParam(name = "page", dataType = "integer", paramType = "query",
+			value = "Results page you want to retrieve (0..N)"),
+		@ApiImplicitParam(name = "size", dataType = "integer", paramType = "query",
+			value = "Number of records per page.")
+	})
+	@ApiOperation(value = "List locations", notes = "Get a list of locations filter by types, favorites, abbreviations and location name.")
 	@RequestMapping(value = "/crops/{cropname}/locations", method = RequestMethod.GET)
 	@ResponseBody
-	public ResponseEntity<List<LocationDto>> listFavoriteLocations(
+	public ResponseEntity<List<LocationDto>> listLocations(
 		@PathVariable final String cropname,
 		@RequestParam(required = false) final String programUUID,
 		@ApiParam(value = "list of location types")
 		@RequestParam(required = false) final Set<Integer> locationTypes,
 		@ApiParam(value = "isFavoriteLocation", required = true)
-		@RequestParam final boolean favoriteLocations) {
+		@RequestParam final boolean favoriteLocations,
+		@RequestParam(required = false) final String name,
+		@ApiIgnore @PageableDefault(page = 0, size = PagedResult.DEFAULT_PAGE_SIZE) final Pageable pageable) {
```

Comment:
Do we need to update [graphical queries ](https://github.com/IntegratedBreedingPlatform/Workbench/blob/d5a13aad5654caa37fcb9952d1021cabc3073780/src/main/web/src/pages/BrAPI-Graphical-Queries/main.js#L25)to pass a large page? and there is a location service in inventory manager too

Reply:
Yup, I need to update this so that if page parameters are not supplied, it will return all filtered records.